File: package/main/src/Date/isSameDay.ts
/**
 * Returns true when two dates represent the same calendar day in local time.
 * @param {Date} left - First date
 * @param {Date} right - Second date
 * @returns {boolean} True if same year/month/day in local time
 * @example
 * isSameDay(new Date("2025-04-15T01:00"), new Date("2025-04-15T23:00")); // true
 * isSameDay(new Date("2025-04-15"), new Date("2025-04-16")); // false
 */
export const isSameDay = (left: Date, right: Date): boolean =>
  left.getFullYear() === right.getFullYear() &&
  left.getMonth() === right.getMonth() &&
  left.getDate() === right.getDate();

File: package/umt_python/src/date/is_same_day.py
from datetime import datetime


def is_same_day(left: datetime, right: datetime) -> bool:
    """
    Return True when two dates represent the same calendar day in local time.

    Args:
        left: First date
        right: Second date

    Returns:
        True if same year/month/day in local time

    Example:
        >>> from datetime import datetime
        >>> is_same_day(datetime(2025, 4, 15, 1, 0), datetime(2025, 4, 15, 23, 0))
        True
        >>> is_same_day(datetime(2025, 4, 15), datetime(2025, 4, 16))
        False
    """
    return (
        left.year == right.year and left.month == right.month and left.day == right.day
    )
